assignments/array-methods.js

```diff
@@ -56,28 +56,63 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(runner){
+    fullName.push(`${runner.first_name}, ${runner.last_name}`);//IF THIS IS CORRECT I CANT BELIEVE I FIGURED IT OUT ON THE FIRST TRY!!!!!
+})
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+const makeThemUpper = runners.map(function(runner){
+    allCaps.push(runner.first_name.toUpperCase())
+})
```

Comment:
This works, but you're not using `map` to the fullest. The way you're using it here is essentially the same as `forEach`. 

"How do I use `map` to the fullest?", you may ask. Here's a thing about `map`: the array method _returns a new array_, and its values are determined by the callback function you pass into it.

In other words, it does a lot of what you did in Challenge 1 for you:

* It creates a new array
* It calls the callback function on each element on the old array
* It takes the result of the callback function and pushes it into the new array
* It then returns you the new array

All that with just one method. Given that, we could rewrite this into:

```js
const allCaps = runners.map(function(runner) {
    return runner.first_name.toUpperCase();
});
```

If you log to the console `allCaps`, you'll find you have the same result.